### PR TITLE
Elevator branch

### PR DESCRIPTION
**Pull request by yuvrajzala26**
Added implementation to go up and down. Just need to edit

### CODE REVIEW

File: src/ArrivalEvent.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import java.time.LocalTime;
+
+public class ArrivalEvent() {
```
Comment (AashnaNarang) [inline src/ArrivalEvent.java]: the spacing looks off, is everything indented normally in eclipse?

File: src/ArrivalEvent.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  int currentFloor;
+  LocalTime time;
+  Elevator elevator;
+
+  public ArrivalEvent(int currentFloor, LocalTime time, Elevator elevator) {
```
Comment (AashnaNarang) [inline src/ArrivalEvent.java]: @danihashweh @abdol99 you guys should all decide what time/date object you want to use to stay consistent

File: src/Direction.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
 
 public enum Direction {
-	UP, DOWN; 
+	UP, DOWN, ArrivalSTOP;
```
Comment (AashnaNarang) [inline src/Direction.java]: why arrivalStop needed?

File: src/Elevator.java [hunk 4]
```diff
@@ -1,4 +1,44 @@
...
+  private Middleman middleman;
+  private Direction direction;
+  private ArrayList<ElevatorButton> buttons;
 
+  public Elevator(Middleman middleman) {
+    this.middleman = middleman;
+    isStationary = true;
+    currentFloor = 0;
+    direction = Direction.UP;
+    buttons = new ArrayList<ElevatorButton>();
```
Comment (AashnaNarang) [inline src/Elevator.java:17]: the constructor should also initialize all of the elevator button objects and add them to the buttons arraylist

File: src/Elevator.java [hunk 5]
```diff
@@ -1,4 +1,44 @@
...
+
+    //Create an array of elevator buttons
```
Comment (AashnaNarang) [inline src/Elevator.java]: remove the // comment, I've learned it's not good practice to add inline comments unless the logic is difficult to understand or you need a reminder for something

File: src/Elevator.java [hunk 6]
```diff
@@ -1,4 +1,44 @@
...
+    for(int i = 0;i < 10;i++) {
+      buttons.add(new ElevatorButton());
```
Comment (AashnaNarang) [inline src/Elevator.java:29]: why are there 10 elevator buttons? Elevator constructor should probably take in a "numFloors" variable to know how many to create

File: src/Elevator.java [hunk 7]
```diff
@@ -1,4 +1,44 @@
...
+    }
+  }
+
+  /*public void clickButton(int button) {
+    buttons.get(button).click();
+
```
Comment (AashnaNarang) [inline src/Elevator.java]: remove these comments

File: src/ElevatorButton.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+public class ElevatorButton {
+  FloorLamp lamp;
+
+  public ElevatorButton() {
+    lamp = new FloorLamp(false);
```
Comment (AashnaNarang) [inline src/ElevatorButton.java]: If you guys are also using FloorLamp then the class should be renamed to "ButtonLamp"

File: src/Elevator.java [hunk 9]
```diff
@@ -1,4 +1,44 @@
...
+    scheduler.putEvent(new ElevatorEvent(this.currentFloor, button, new Time(), this);
+  }*/
+
+  public void run() {
+    while(true) {
+        FloorEvent floorevent = middleman.getFloorEvent();
+        if(floorevent != null) {
+          int floors = floorevent.getDestination() - floorevent.getSource();
+          currentFloor += floors;
+
```
Comment (AashnaNarang) [inline src/Elevator.java:49]: The code for turning elevator buttons/lamps on/off is missing

File: src/Elevator.java [hunk 10]
```diff
@@ -1,4 +1,44 @@
...
+    //TODO Time, how long?
+    scheduler.putEvent(new ElevatorEvent(this.currentFloor, button, new Time(), this);
+  }*/
+
+  public void run() {
+    while(true) {
+        FloorEvent floorevent = middleman.getFloorEvent();
+        if(floorevent != null) {
+          int floors = floorevent.getDestination() - floorevent.getSource();
+          currentFloor += floors;
```
Comment (AashnaNarang) [inline src/Elevator.java]: We should be sending arrival events at every floor you go to. I recommend doing something like `
```suggestion
          if (i=currentFloor; i <= floors; i++ ) { 
              currentFloor += i;
              //create arrival event
              // put event in middle man object
           }
```

File: src/Elevator.java [hunk 11]
```diff
@@ -1,4 +1,44 @@
...
+
+          ArrivalEvent arrivalEvent = new ArrivalEvent(this.currentFloor, LocalTime.now(), this);
+          middleman.putArrivalEvent();
```
Comment (AashnaNarang) [inline src/Elevator.java]: add arrival event as an argument to putArrivalEvent()

File: src/ElevatorButton.java [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+  }
+
+  public void click() {
+    if(lamp.isOn()) {
+      lamp.setOn(false);
```
Comment (AashnaNarang) [inline src/ElevatorButton.java]: you can simplify the logic in this method by doing 
```suggestion
      lamp.setOn(!lamp.isOn());
```

File: src/Direction.java (hunk 3 above)
Comment (AashnaNarang) [reply src/Direction.java]: I don't see it used anywhere, can you remove it

File: src/ArrivalEvent.java (hunk 2 above)
Comment (abdol99) [reply src/ArrivalEvent.java]: We use Localtime as well

File: src/Elevator.java [hunk 13]
```diff
@@ -1,4 +1,44 @@
+import java.lang.Math;
+import java.time.LocalTime;
+import java.util.ArrayList;
 
-public class Elevator {
+public class Elevator implements Runnable {
+  private int currentFloor;
+  private boolean isStationary;
+  private Middleman middleman;
```
Comment (AashnaNarang) [inline src/Elevator.java]: should be `private MiddleMan middleMan`

File: src/Elevator.java [hunk 14]
```diff
@@ -1,4 +1,44 @@
...
+  /*public void clickButton(int button) {
+    buttons.get(button).click();
+
+    //TODO Time, how long?
+    scheduler.putEvent(new ElevatorEvent(this.currentFloor, button, new Time(), this);
+  }*/
+
+  public void run() {
+    while(true) {
+        FloorEvent floorevent = middleman.getFloorEvent();
```
Comment (AashnaNarang) [inline src/Elevator.java]: use camelcase for variable names, call this floorEvent

**Comment by AashnaNarang** [comment]
Outdated

**Pull request closed without merging**